forge/src/gas_report.rs

```diff
@@ -0,0 +1,147 @@
+
+        if trace.address == *CHEATCODE_ADDRESS || trace.address == *HARDHAT_CONSOLE_ADDRESS {
+            return
+        }
+
+        if let Some(name) = &trace.contract {
+            let report_for = self.report_for.iter().any(|s| s == name);
+            if report_for || report_for_all {
+                let mut contract_report =
+                    self.contracts.entry(name.to_string()).or_insert_with(Default::default);
+
+                match &trace.data {
+                    RawOrDecodedCall::Raw(bytes) if trace.created => {
+                        contract_report.gas = trace.gas_cost.into();
+                        contract_report.size = bytes.len().into();
+                    }
+                    // TODO: More robust test contract filtering
+                    RawOrDecodedCall::Decoded(func, _) if !func.starts_with("test") => {
+                        let function_report = contract_report
+                            .functions
+                            .entry(func.clone())
+                            .or_insert_with(Default::default);
+                        function_report.calls.push(trace.gas_cost.into());
+                    }
+                    _ => (),
+                }
+            }
+        }
+
+        node.children.iter().for_each(|index| {
+            self.analyze_node(*index, arena, report_for_all);
+        });
+    }
+
+    #[must_use]
+    pub fn finalize(mut self) -> Self {
+        self.contracts.iter_mut().for_each(|(_, contract)| {
+            contract.functions.iter_mut().for_each(|(_, func)| {
+                func.calls.sort();
+                func.min = func.calls.first().cloned().unwrap_or_default();
+                func.max = func.calls.last().cloned().unwrap_or_default();
+                func.mean =
+                    func.calls.iter().fold(U256::zero(), |acc, x| acc + x) / func.calls.len();
+
+                let len = func.calls.len();
+                func.median = if len > 0 {
+                    if len % 2 == 0 {
+                        (func.calls[len / 2 - 1] + func.calls[len / 2]) / 2
+                    } else {
+                        func.calls[len / 2]
+                    }
+                } else {
+                    0.into()
+                };
+            });
+        });
+        self
+    }
+}
+
+impl Display for GasReport {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+        for (name, contract) in self.contracts.iter() {
+            let mut table = Table::new();
+            table.load_preset(UTF8_FULL).apply_modifier(UTF8_ROUND_CORNERS);
+            table.set_header(vec![Cell::new(format!("{} contract", name))
+                .add_attribute(Attribute::Bold)
+                .fg(Color::Green)]);
+            table.add_row(vec![
+                Cell::new("Deployment Cost").add_attribute(Attribute::Bold).fg(Color::Cyan),
+                Cell::new("Deployment Size").add_attribute(Attribute::Bold).fg(Color::Cyan),
```

Comment:
Should be possible without a trait since `.color` takes `self` and returns `Self`, so something like

```rust
struct CellColorizer {
  should_color: bool
}

impl CellColorizer {
  fn color(cell: Cell, color: ...) -> Cell {
    if self.should_color {
      cell.color(color)
    } else {
      cell
    }
  }
}

let colorizer = CellColorizer::new(self.stdout);

colorizer.color(Cell::new("Function Name"), Color::Magenta)
```

Same for attributes like bold etc. (maybe `CellFormatter` is a better name)